docs/design/phases.md

```diff
@@ -0,0 +1,127 @@
+## Phases in kubeadm
+
+`kubeadm init` does a lot. In order to make kubeadm the common building block for Kubernetes, each logical phase kubeadm does during `init` should be invokable separately. 
+Here's an explanation what kubeadm does in these different phases:
+
+#### A note on constants / well-known values and paths
+
+We have to draw the line somewhere about what should be configurable, what shouldn't, and what should be hard-coded in the binary.
+
+We've decided to make the Kubernetes directory /etc/kubernetes a constant in the application, since it is clearly the given path in a majority of cases,
+and the most intuitive location. Having that path configurable would confuse readers of an on-top-of-kubeadm-implemented deployment solution.
+
+Further, having the admin/kubelet kubeconfig path as constants as well (`/etc/kubernetes/admin.conf` and `/etc/kubernetes/kubelet.conf`) would make things easier as well.
+Is it ok to have `/etc/kubernetes/kubelet.conf` as a "standard" between official deployments? It defines the path for the kubelet's KubeConfig file
+
+### Certificates
+
+First off, kubeadm generates certificate and private key pairs for different purposes.
+Certificates are stored by default in `/etc/kubernetes/pki`.
+
+There should be:
```

Comment:
There should also be a CA keypair and client certificate for the Authenticating Proxy described here: https://kubernetes.io/docs/admin/authentication/#authenticating-proxy.  It's near zero cost in the kube-apiserver to have it, there's no exposure unless actual machine access is had, and future extension stories (https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/kubernetes-sig-api-machinery/o2v-HjAHk0k/Rg4lNeEACQAJ) rely upon it.